### ISSUE TITLE
Asserting flash message on `redirect`.

### ISSUE THREAD

**Issue by sodapopcan** [germsvel/phoenix_test#154]
This may be related to #146.

We were trying to convert phx_gen_auth tests to PhoenixTest and got stuck on flash messages.  A `redirect` is used to in some cases which the generated tests check the conn directly:

```
assert Phoenix.Flash.get(conn.assigns.flash, :info) =~ "If your email is in our system"
```

This isn't possible in PhoenixTest as the returned conn's flash key is empty.

Is there a clean way around this I'm missing or would it require a PR?

**Comment by germsvel** [comment]
Interesting. @sodapopcan I think this happens because we emulate the redirect by performing a visit -- that's the same strategy LiveView does with `follow_redirect`. But... I think LiveView does some extra stuff to keep the flash alive. 

I've run into this in other work, so I have a sense of what _might_ be going on. But just to be sure, can you confirm what kind of redirect you're performing? I assume (by your excellent use of syntax highlighting) it's a full `redirect` (not a `navigate`), but just want to confirm. 

Can you also let me know what happens before the redirect? Is it a form submission? Is it something else? 

Basically, any info that can help us write a test to replicate the error would be excellent.

**Comment by sodapopcan** [comment]
Yes, sorry I should have used the slash notation to be more clear I was talking about a function (I noticed José not bothering with the slash so I gave up on it long ago 😅).  Yes, I'm specifically talking about `redirect/2`.  Any other kind of navigation works.  I've never actually used `redirect/2` and I'm not 100% certain on why it's used in phx_gen_auth.  I'm assuming it's a security thing though in its docs, the sole example is redirecting to an external site.  So I hadn't run into this yet and only am now since this is the first time I've tried converting the generate auth tests to PhoenixTest.

**Comment by germsvel** [comment]
@sodapopcan no worries on the slash vs no-slash. I do it without the slash all the time too (sometimes the functions are the same with different arities). Just wanted to make sure I was following what you were saying. 

Yeah, this is something we need to fix. I'll try to get to it when I can, but if you're up for it, I'd also be happy to review a PR. There is some trickiness to this one though, so you'd have to figure out how to push flashes when redirecting so that tests can pick them up.

**Comment by germsvel** [comment]
@sodapopcan I think I figured out the issues and fixed them in https://github.com/germsvel/phoenix_test/commit/ca7edf5a97187839d5116972a24115367cef7d8d. Can you give that a try at some point to see if that fixes things? 

Note: I've tested things through the `assert_has/3` helper (e.g. `assert_has(session, "#flash-group", text: "flash message")`), not directly through the `Flash.get/2` like you showed in your example, but hopefully it still works.
